halo-dev/halo
# feat: Add rule resolver and authorization filter

### What this PR does?
- 新增规则解析器
- 新增鉴权过滤器
- 增加集成测试
see also #1895

### Why we need it?
需要对请求进行鉴权

### how to test it?
1. 登录获取 token
```shell
curl -X POST http://localhost:8090/api/v1/oauth2/token -H "Content-Type: application/x-www-form-urlencoded" -d 'username=user&password=123456&grant_type=password'
```
2. 使用获取到的 access_token 请求测试资源(替换如下的 your_access_token)，预期结果为：`Now you see me.`
```
curl http://localhost:8090/api/v1/posts -H "Authorization: Bearer your_access_token"
```
3. 请求没有权限的资源得到 403 页面
```
curl http://localhost:8090/api/v1/tags -H "Authorization: Bearer your_access_token"
```

/kind feature
/area core
/milestone 2.0
/cc @halo-dev/sig-halo
